packages/cli/src/helpParamGenerator.ts

```diff
@@ -19,21 +19,27 @@ import {
   MultiSelectQuestion,
 } from "@microsoft/teamsfx-api";
 
-import { FxCore } from "@microsoft/teamsfx-core";
+import { FxCore, isMultiEnvEnabled } from "@microsoft/teamsfx-core";
 import AzureAccountManager from "./commonlib/azureLogin";
 import AppStudioTokenProvider from "./commonlib/appStudioLogin";
 import GraphTokenProvider from "./commonlib/graphLogin";
 import CLILogProvider from "./commonlib/log";
 import CLIUIInstance from "./userInteraction";
 import { flattenNodes, getSingleOptionString, toYargsOptions } from "./utils";
 import { Options } from "yargs";
-import { RootFolderNode, sqlPasswordConfirmQuestionName } from "./constants";
+import {
+  EnvNode,
+  EnvNodeNoCreate,
+  RootFolderNode,
+  sqlPasswordConfirmQuestionName,
+} from "./constants";
 import { NoInitializedHelpGenerator } from "./error";
 
 export class HelpParamGenerator {
   private core: FxCore;
   private questionsMap: Map<string, QTreeNode> = new Map<string, QTreeNode>();
   private initialized = false;
+  private static showEnvStage: string[] = [Stage.build, Stage.publish, Stage.deploy];
```

Comment:
we also need to support validate the application of one specified environment.

Reply:
Will add later.